[Quant][Inductor] Enable quantization conv pattern fusion inside inductor

leslie-fang-intel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107951
* #106958
* #106838
* #106836
* #107565
* #105996
* #105906
* #105639
* #105456
* #105455
* #104590
* __->__ #104588
* #104581
* #104580

**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

leslie-fang-intel committed:
enable quantization conv pattern fusion inside inductor

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"




cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

kimishpatel commented on torch/_inductor/fx_passes/quantization.py:
```diff
@@ -143,8 +143,109 @@ def qconv(match: Match, *args, **kwargs):
+
 def register_quantization_lowerings():
     _register_quantized_conv_lowering(quantize_conv_output_pattern)
+    _register_quantized_conv_lowering_pt2e(quantize_conv_output_pattern_pt2e)
```
why does pt2e pattern have packed weight instead of dequantized weight and then aten.convolution?

kimishpatel replied:
Also writing these pattern manually also seems to rely on the decomposition rules and how the op will appear in the exported graph. Is that a valid assumption?

jgong5 replied:
> why does pt2e pattern have packed weight instead of dequantized weight and then aten.convolution?

Weight prepacking is done earlier in inductor as part of the constant folding pass in freezing graph pass where weights are marked as constants and folded with "prepacking op" into the prepacked weights. We don't create new constant tensors in the graph fusion passes.

> Also writing these pattern manually also seems to rely on the decomposition rules and how the op will appear in the exported graph. Is that a valid assumption?

That's a good question. I guess it is a nature of pattern-based approach - we have to make assumptions about how the graph would look like. I think that should be fine since the decomposition rules are also what PyTorch core maintains.

kimishpatel commented on torch/_inductor/ir.py:
```diff
@@ -4234,6 +4234,172 @@ def create(
+            dilation = const_args[3]
+            groups = const_args[4]
+
+        self.kernel = "torch.ops.quantized.qconv2d_pt2e"
```
So this is just dispatching to a handwritten kernel, rather than using compiler to do codegen? Why is it done via inductor?

Also as I said in another PR, we should really use backend specific ops and custom libs for this purpose.

We used to do torch.ops.quantized.* because we effectively had "quantized" backend in aten. But quantized model in PT2E is represented via canonical aten ops (except for chose_qparams). Thus if you take "dq -> conv -> q" and convert that pattern to a custom op "quantized_conv", then such an op should not and does not need to be in "torch.op.quantized.*". It should really be in "torch.ops.onednn" as it is backend specific. cc: @jerryzh168

jerryzh168 replied:
yeah agree, I think this should be put into the `torch.ops.onednn` namespace

jgong5 replied:
> So this is just dispatching to a handwritten kernel, rather than using compiler to do codegen? Why is it done via inductor?

Not all kernels in the inductor compilation path are codegen. There are both library-based fusion kernels and codegen kernels. The former is primarily for fusions with conv and gemm ops and the latter is for memory intensive operations like pointwise and reduction.

> yeah agree, I think this should be put into the `torch.ops.onednn` namespace

Sure, let's put them under onednn namespace.

leslie-fang-intel replied:
Thanks and moved them into onednn namespace.

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

kimishpatel commented on aten/src/ATen/native/quantized/cpu/qconv.cpp:
```diff
@@ -1780,6 +1780,9 @@ class QConvoneDNN final {
         ".")
     }
+
+    output_scale = 1.0 / output_scale;
```
should probably rename this to inv_output_scale

leslie-fang-intel replied:
Thanks and changed.

kimishpatel reviewed (commented):
Looks good to me but for inductor stuff I will let other folks take care of the review.

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel commented on torch/_inductor/fx_passes/quantization.py:
```diff
@@ -96,74 +104,77 @@
+        pattern_match_count += 1
+        print(
+            "---- matched the pattern ----: {}".format(pattern_match_count), flush=True
+        )
```
Thanks and removed.

leslie-fang-intel commented:
Thanks for the comments @jgong5 @jerryzh168 @kimishpatel. Please help to take a look again.

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

jgong5 commented on test/inductor/test_mkldnn_pattern_matcher.py:
```diff
@@ -389,11 +389,13 @@ def forward(self, x):
                 (v,),
-                3,
-                10 if auto_insert_channel_last_node else 9,
+                4,
+                17 if auto_insert_channel_last_node else 16,
```
Is the comment aligned with the changes of count?

leslie-fang-intel replied:
Sorry for the mistake, fixed the comments.

jgong5 reviewed (approved):
LGTM in general with a minor comment on the code comment.

leslie-fang-intel commented:
Hi @eellison @jansel, could you also kindly help to take a look of this PR?

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

eellison reviewed (approved):
LGTM

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel committed:
Update on "[Quant][Inductor] Enable quantization conv pattern fusion inside inductor"


**Summary**
Enable the `dequant-quantization-quant` pattern fusion and lowering inside inductor. 

**Test Plan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d_unary
```

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 voznesenskym penguinwu EikanWang Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

leslie-fang-intel commented:
@pytorchbot merge
